2024Project-main/2024FinalYearProject/Controllers/ClientController.cs

```diff
@@ -23,29 +23,54 @@ public ClientController(IRepositoryWrapper repo, UserManager<AppUser> userManage
             _userManager = userManager;
         }
 
-        public IActionResult Index()
+        public async Task<IActionResult> IndexAsync()
         {
-            return View();
+
+
+            var username = User.Identity.Name;
+
+            var user = await _userManager.FindByNameAsync(username);
+            var model = new UpdateProfileViewModel
+            {
+                Email = user.Email,
+                PhoneNumber = user.PhoneNumber,
+
+                IDNumber = user.IDnumber,
+
+                Userrole = user.UserRole,
+                Lastname = user.LastName + " " + user.FirstName,
+
+            };
+            return View(model);
+           
         }
 
         // Get Notification
         public async Task<IActionResult> NotificationMessage()
         {
             var username = User.Identity.Name;
-
             var user = await _userManager.FindByNameAsync(username);
 
             if (user == null)
             {
-
                 return NotFound();
             }
-            var allNotifications = await _repo.Notification.GetAllAsync();
 
+            var allNotifications = await _repo.Notification.GetAllAsync();
             var userNotifications = allNotifications.Where(n => n.AppUserId == user.Id).ToList();
 
+            foreach (var notification in userNotifications)
+            {
+                if (!notification.IsRead)
+                {
+                    notification.IsRead = true;
+                    await _repo.Notification.UpdateAsync(notification);
```

Comment:
Just a question, does this work the same way as `_repo.SaveChanges();`

Reply:
yes man each contain update and save as _repo.SaveChanges();